ios/browser/api/history/brave_history_observer.h

```diff
@@ -0,0 +1,55 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_HISTORY_BRAVE_HISTORY_OBSERVER_H_
+#define BRAVE_IOS_BROWSER_API_HISTORY_BRAVE_HISTORY_OBSERVER_H_
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class IOSHistoryNode;
+
+OBJC_EXPORT
+@protocol HistoryServiceObserver <NSObject>
+@optional
+
+/// Notifying when `history_service` has finished loading.
+- (void)historyServiceLoaded;
+
+/// Notifying when `history_service` is being deleted.
+- (void)historyServiceBeingDeleted;
+
+/// Observing when user visits an URL.
+/// @param historyNode - History object which has been visited
+- (void)historyNodeVisited:(IOSHistoryNode* _Nonnull)historyNode;
+
+/// Observing when a URL is added or modified
+/// @param historyNodeList - History object list which has been modified
+- (void)historyNodesModified:(NSArray<IOSHistoryNode*>*)historyNodeList;
+
+/// Observing when one or more URLs are deleted
+/// @param historyNodeList - History object list which has been deleted
+///                          historyNodeList will be empty is all history is
+///                          deleted
+/// @param isAllHistory - Boolean that notifies all history is deleted
+- (void)historyNodesDeleted:(NSArray<IOSHistoryNode*>*)historyNodeList
+               isAllHistory:(bool)isAllHistory;
+@end
+
+OBJC_EXPORT
+@protocol HistoryServiceListener
+- (void)destroy;
+@end
+
+OBJC_EXPORT
+@interface HistoryServiceListenerImpl : NSObject <HistoryServiceListener>
+- (instancetype)init:(id<HistoryServiceObserver>)observer
+      historyService:(void*)service;
+@end
```

Comment:
Shouldn't be public

Reply:
Fix at https://github.com/brave/brave-core/pull/8869/commits/413969be02cd494e51fa9cd4a61bf5beaa1bc8c7